app/src/hooks/useStakingOptions.ts

```diff
@@ -116,24 +116,28 @@ const majors = [WBTCPO, TBTC, WSTETHPO, RETHPO, WETHPO, WSOL];
 
 const BP = 0.01 / 100;
 
+type FeeBasedOnSoParams = Pick<SOState, 'baseMint' | 'quoteMint' | 'authority' | 'soName'>;
 /**
  * Utility function that returns fee multiplier for exercising options,
  * based on https://github.com/Dual-Finance/staking-options/blob/b902c46e0ea78fdf7edf42967b1583c74b995743/programs/staking-options/src/common.rs#L88C18-L88C18
  * */
-function feeByPair(base: PublicKey, quote: PublicKey, authority: PublicKey): number {
-  if (authority.toString() === DUAL_DAO_WALLET_PK.toString()) {
+function getFeeBasedOnSO({ baseMint, quoteMint, authority, soName }: FeeBasedOnSoParams): number {
+  if (
+    authority.toString() === DUAL_DAO_WALLET_PK.toString() ||
+    (baseMint.toString() === Config.dualMintPk().toString() && soName.includes('GSO'))
```

Comment:
TODO to remove this code. Should not be just checking mint for GSO. Would be better to check on the state and determine who the ultimate owner of the GSO is and ignore if that is the DAO.